src/main/java/co/cask/hydrator/plugin/FastFilter.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Macro;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.Emitter;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.Transform;
+import co.cask.cdap.etl.api.TransformContext;
+import com.google.common.annotations.VisibleForTesting;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * Quickly filters records based on a set criteria.
+ */
+@Plugin(type = Transform.PLUGIN_TYPE)
+@Name("FastFilter")
+@Description("Only allows records through that pass the specified criteria.")
+public final class FastFilter extends Transform<StructuredRecord, StructuredRecord> {
+  private final Config config;
+  private Pattern regexPattern;
+
+  @VisibleForTesting
+  public FastFilter(Config config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+    super.configurePipeline(pipelineConfigurer);
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    config.validate(inputSchema);
+    // Just checking that the operator is valid
+    shouldAllowThrough("", config.operator, "");
+    pipelineConfigurer.getStageConfigurer().setOutputSchema(pipelineConfigurer.getStageConfigurer().getInputSchema());
+  }
+
+  @Override
+  public void initialize(TransformContext context) throws Exception {
+    super.initialize(context);
+    if (config.operator.contains("regex")) {
+      regexPattern = Pattern.compile(config.criteria);
+    }
+  }
+
+  @Override
+  public void transform(StructuredRecord in, Emitter<StructuredRecord> emitter) throws Exception {
+    if (in.get(config.sourceField) != null) {
+      Schema inputSchema = in.getSchema();
+      if (!inputSchema.getField(config.sourceField).getSchema().isSimpleOrNullableSimple()) {
```

Comment:
Should be checked during configure pipeline, would be frustrating for the user to be able to publish the pipeline and then finding out that schema is wrong. If we are already checking there, we might not need to check it again.